test/fiber/test_address_resolve.rb

```diff
@@ -0,0 +1,278 @@
+# frozen_string_literal: true
```

Comment:
I'd like to add a test for every ruby method that touches `rb_schedule_getaddrinfo`.
But this file is already too big so  not sure if that's a good idea.

Reply:
I understand how you feel. I don't have any firm advice about it, except that we want to try and pin down the behaviour as much as "makes sense". Ideally, tests represent the extent of what people would do with the interface, so if we break something, it's picked up by tests before users start complaining :)